Added Tailwind and shadcn and edited Navbar

AnnieMcMahon opened the pull request:
- Added Tailwind and shadcn to the project
- Edited the Navbar to be responsive and better formatted, using shadcn components
- Made other minor adjustments to css file until we transition each component to Tailwind (after adding Tailwind, the forms didn't look the same)
- Deleted already implemented action items from the About page and tweaked the links

AnnieMcMahon committed:
added Tailwind and edited Navbar

AnnieMcMahon committed:
added green color to AllTrails links

tbmeans reviewed (approved):
app/(dashboard)/about/page.jsx
app/ui/components/Navbar.jsx
components/ui/button.jsx
components/ui/menubar.jsx
components/ui/navigation-menu.jsx
* Edited the Navbar to be responsive and better formatted, using shadcn components
* Deleted already implemented action items from the About page and tweaked the links

Responsive navbar is now a success! (e.g. items no longer overflow for narrow widths) and I confirm removal of done action items. Noted/approved removal of "Navbar.css" import and it's required "usePathname," all replaced by importing shadcdn components.
Noted/approve the code to import all other shadcn components.

tbmeans reviewed (changes requested):
app/global.css
* Added Tailwind and shadcn to the project
* Made other minor adjustments to css file until we transition each component to Tailwind (after adding Tailwind, the forms didn't look the same)

Approved:
--I like the removal of traditional underline/blue/purple style from the links.
--I like the font change from TNR to Helvetica sans-serif.
--The whitespace between list headings and list items is improved.
--In narrow windows I can't tell difference between Join Hike <p> paddings but the added rule doesn't hurt anything.
--The Home page instructions' list-item style additions are fine.

Issues:
--On the Home page I'm not finding where the textarea element is even used. I see that the box of instructions is a class "text-box" but there is no literal textarea element that I can find.
--When you view the home page in narrow/mobile widths, the instructions list bullets are about to overflow the border. A fix for this is to give the instructions' unordered-list tag a margin-left of 0.5rem (giving margin back to the list-item has no effect). Don't want too much rem or it'll make the list text wrap too much.

tbmeans reviewed (approved):
app/ui/components/ChosenTrail.jsx
app/ui/components/HikePost.jsx
app/ui/components/TrailPost.jsx
I confirm/approve correction of term from "ft" to "ft elev gain".
Bravo for the use of Tailwind class to re-style the "AllTrails Link"!

tbmeans reviewed (changes requested):
app/ui/forms/EditBioForm.jsx
Suggestion for the Tailwind classes added to Avatar's input element:
to keep "No File Chosen" from truncating to "No...sen" for narrow views, change file:mr-5 to file:mr-6 and 
file:text-xs to file:text-md, but the downside is that it puts "No File Chosen" very close to the Choose File button.

AnnieMcMahon committed:
fixed home page bullet styling and avatar change styling

AnnieMcMahon commented:
Thank you for your review, Tim! I fixed the styling for the home page bullets and the Avatar change section. Let me know if it's better now. You are right, the textarea is not used on the Home page, but I added the styling to the global.css file because it applies to all the forms. It's easier than adding it to each individual form. This might change if we use shadcn's Form component in the future.
